src/v/iceberg/manifest_entry_values.cc

```diff
@@ -0,0 +1,244 @@
+        return data_file_content_type::equality_deletes;
+    }
+    throw std::invalid_argument(
+      fmt::format("Invalid data file content type: {}", c));
+}
+
+iobuf format_to_str(data_file_format f) {
+    switch (f) {
+    case data_file_format::avro:
+        return iobuf::from("avro");
+    case data_file_format::orc:
+        return iobuf::from("orc");
+    case data_file_format::parquet:
+        return iobuf::from("parquet");
+    }
+}
+
+data_file_format format_from_str(std::string_view s) {
+    return string_switch<data_file_format>(s)
+      .match("avro", data_file_format::avro)
+      .match("AVRO", data_file_format::avro)
+      .match("orc", data_file_format::orc)
+      .match("ORC", data_file_format::orc)
+      .match("parquet", data_file_format::parquet)
+      .match("PARQUET", data_file_format::parquet);
+}
+
+std::unique_ptr<struct_value> data_file_to_value(const data_file& file) {
+    auto ret = std::make_unique<struct_value>();
+    ret->fields.emplace_back(int_value(content_to_int(file.content_type)));
+    ret->fields.emplace_back(string_value(iobuf::from(file.file_path)));
+    ret->fields.emplace_back(string_value(format_to_str(file.file_format)));
+    ret->fields.emplace_back(std::move(file.partition.copy().val));
+    ret->fields.emplace_back(
+      long_value(static_cast<int64_t>(file.record_count)));
+    ret->fields.emplace_back(
+      long_value(static_cast<int64_t>(file.file_size_bytes)));
+
+    // TODO: serialize the rest of the optional fields.
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    ret->fields.emplace_back(std::nullopt);
+    return ret;
+}
+
+data_file data_file_from_value(struct_value v) {
+    data_file file;
+    auto& fs = v.fields;
+    if (fs.size() < 10) {
+        throw std::invalid_argument("Expected more values");
+    }
+    file.content_type = content_from_int(
+      get_required_primitive<int_value>(std::move(fs[0])));
+    file.file_path = from_iobuf(
+      get_required_primitive<string_value>(std::move(fs[1])));
+    file.file_format = format_from_str(
+      from_iobuf(get_required_primitive<string_value>(std::move(fs[2]))));
+    file.partition = {get_required_struct(std::move(fs[3]))};
+    file.record_count = get_required_primitive<long_value>(std::move(fs[4]));
+    file.file_size_bytes = get_required_primitive<long_value>(std::move(fs[5]));
+    file.column_sizes = get_counts_map(std::move(fs[6]));
+    file.value_counts = get_counts_map(std::move(fs[7]));
+    file.null_value_counts = get_counts_map(std::move(fs[8]));
+    file.nan_value_counts = get_counts_map(std::move(fs[9]));
+    return file;
```

Comment:
Don't love this, open to feedback to improve

Reply:
Yeah, I just opened this and data_file_type in a split so I could double check them by hand. Since Avro is not self-describing this is sensitive to ordering :-(

I don't think this is quite right. This should work if the writer used the same schema that we're reading with, but not otherwise. That's probably not so bad for now because we're the writer and the reader, but if we change the order that we write fields, or start writing out that `distinct_counts` field this will break.

I believe the Avro way to do this would be to pass the schema that the writer used to this method (it should be in the file's header) and match the fields using that.

Reply:
To be clear, I'm ok with leaving it like this for now, but maybe add a todo for it.

Reply:
I don't believe this is true. We pass in a schema to the Avro reader, and IIUC the Avro reader reconciles that schema with the writer schema.

Can look for a more exact link later, but for instance, this mentions reconciling differences between the reader and writer schema https://github.com/redpanda-data/avro/blob/1410e79f9df61669c2d52f6d0643e6c35156e615/lang/c%2B%2B/api/Decoder.hh#L179-L191

As an example, this diff continues to successfully read the fixed manifest that's checked into the tree https://gist.github.com/andrwng/8d695b3b15260ef86d7589272fe32c84